NitroxPatcher/Patches/Dynamic/StoryGoalScheduler_Schedule_Patch.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using HarmonyLib;
+using NitroxClient.Communication.Abstract;
+using NitroxModel.Helper;
+using NitroxModel.Packets;
+using Story;
+
+namespace NitroxPatcher.Patches.Dynamic
+{
+    public class StoryGoalScheduler_Schedule_Patch : NitroxPatch, IDynamicPatch
+    {
+        public static readonly Type TARGET_CLASS = typeof(StoryGoalScheduler);
+        public static readonly MethodInfo TARGET_METHOD = Reflect.Method((StoryGoalScheduler t) => t.Schedule(default(StoryGoal)));
+        private static readonly IPacketSender packetSender = Resolve<IPacketSender>();
+
+        // __state is a bool made to prevent duplicated entries, if it's false, then it should be skipped
+        public static bool Prefix(StoryGoal goal, out bool __state)
+        {
+            __state = StoryGoalScheduler.main.schedule.Any(scheduledGoal => scheduledGoal.goalKey == goal.key)
+                || goal.goalType == Story.GoalType.Radio && StoryGoalManager.main.pendingRadioMessages.Contains(goal.key);
+
+            return !__state;
+        }
+
+
+        public static void Postfix(StoryGoal goal, bool __state)
+        {
+            if (__state || goal.key == "PlayerDiving" || PDALog.entries.ContainsKey(goal.key))
+            {
+                return;
+            }
+
+            ScheduledGoal scheduledGoal = StoryGoalScheduler.main.schedule.Find(scheduledGoal => scheduledGoal.goalKey == goal.key);
```

Comment:
Performance wise I would use the SN calculation of `timeExecute` instead of the linq query for finding the scheduledGoal.